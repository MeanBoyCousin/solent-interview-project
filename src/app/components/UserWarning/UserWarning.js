import React from 'react'

import { promptYInOut } from 'Motion/variants'
import { UserWarningWrapper } from './UserWarning.styled'

const UserWarning = () => {
    return (
        <UserWarningWrapper
            variants={promptYInOut}
            initial="hidden"
            animate="visible"
            exit="exit"
        >
            <p>
                There was a problem fetching the staff profiles. Please refresh
                to try again.
            </p>
        </UserWarningWrapper>
    )
}

export { UserWarning }
