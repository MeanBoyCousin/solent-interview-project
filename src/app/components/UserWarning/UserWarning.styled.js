import styled from 'styled-components'
import { motion } from 'framer-motion'

const UserWarningWrapper = styled(motion.div)`
    position: fixed;
    left: 0;
    bottom: 0;
    width: 100%;
    padding: 5%;
    box-sizing: border-box;
    background: ${props => props.theme['dark-grey']};
    color: white;
    text-align: center;
    font-weight: 600;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
`

export { UserWarningWrapper }
