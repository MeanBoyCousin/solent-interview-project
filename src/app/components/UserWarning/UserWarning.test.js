import React from 'react'
import '@testing-library/jest-dom'
import { render, cleanup } from '@testing-library/react'

import { UserWarning } from './UserWarning'

afterEach(cleanup)

describe('UserWarning component', () => {
    test('should render with correct warning message', () => {
        const { asFragment } = render(<UserWarning />)
        expect(asFragment()).toMatchSnapshot()
    })
})
