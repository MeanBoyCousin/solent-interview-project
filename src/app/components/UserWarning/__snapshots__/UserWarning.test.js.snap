// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserWarning component should render with correct warning message 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa iABdWD"
    style="opacity: 0.2; transform: translateY(100%) translateZ(0);"
  >
    <p>
      There was a problem fetching the staff profiles. Please refresh to try again.
    </p>
  </div>
</DocumentFragment>
`;
